http_handler_test.go

```diff
@@ -22,110 +22,124 @@ package zap_test
-	lvl, _ := newHandler()
-	code, body := makeRequest(t, "PUT", lvl, strings.NewReader(`{`))
-	assertCodeBadRequest(t, code)
-	assertJSONError(t, body)
-}
-
-func TestHTTPHandlerNoLevelSpecified(t *testing.T) {
-	lvl, _ := newHandler()
-	code, body := makeRequest(t, "PUT", lvl, strings.NewReader(`{}`))
-	assertCodeBadRequest(t, code)
-	assertJSONError(t, body)
-}
-
-func TestHTTPHandlerMethodNotAllowed(t *testing.T) {
-	lvl, _ := newHandler()
-	code, body := makeRequest(t, "POST", lvl, strings.NewReader(`{`))
-	assertCodeMethodNotAllowed(t, code)
-	assertJSONError(t, body)
+func TestAtomicLevelServeHTTP(t *testing.T) {
+	tests := map[string]struct {
+		Method        string
+		ContentType   string
+		Body          string
+		ExpectedCode  int
+		ExpectedLevel zapcore.Level
+	}{
+		"GET": {
+			Method:        http.MethodGet,
+			ExpectedCode:  http.StatusOK,
+			ExpectedLevel: zap.InfoLevel,
+		},
+		"PUT JSON": {
+			Method:        http.MethodPut,
+			ExpectedCode:  http.StatusOK,
+			ExpectedLevel: zap.WarnLevel,
+			Body:          `{"level":"warn"}`,
+		},
+		"PUT URL encoded": {
+			Method:        http.MethodPut,
+			ExpectedCode:  http.StatusOK,
+			ExpectedLevel: zap.WarnLevel,
+			ContentType:   "application/x-www-form-urlencoded",
+			Body:          "level=warn",
+		},
+		"PUT JSON unrecognized": {
+			Method:       http.MethodPut,
+			ExpectedCode: http.StatusBadRequest,
+			Body:         `{"level":"unrecognized"}`,
+		},
+		"PUT URL encoded unrecognized": {
+			Method:       http.MethodPut,
+			ExpectedCode: http.StatusBadRequest,
+			ContentType:  "application/x-www-form-urlencoded",
+			Body:         "level=unrecognized",
+		},
+		"PUT JSON malformed": {
+			Method:       http.MethodPut,
+			ExpectedCode: http.StatusBadRequest,
+			Body:         `{"level":"warn`,
+		},
+		"PUT URL encoded malformed": {
+			Method:       http.MethodPut,
+			ExpectedCode: http.StatusBadRequest,
+			ContentType:  "application/x-www-form-urlencoded",
+			Body:         "level=%",
+		},
+		"PUT JSON unspecified": {
+			Method:       http.MethodPut,
+			ExpectedCode: http.StatusBadRequest,
+			Body:         `{}`,
+		},
+		"PUT URL encoded unspecified": {
+			Method:       http.MethodPut,
+			ExpectedCode: http.StatusBadRequest,
+			ContentType:  "application/x-www-form-urlencoded",
+			Body:         "",
+		},
+		"POST JSON": {
+			Method:       http.MethodPost,
+			ExpectedCode: http.StatusMethodNotAllowed,
+			Body:         `{"level":"warn"}`,
+		},
+		"POST URL": {
+			Method:       http.MethodPost,
+			ExpectedCode: http.StatusMethodNotAllowed,
+			ContentType:  "application/x-www-form-urlencoded",
+			Body:         "level=warn",
+		},
+	}
+
+	for name, test := range tests {
+		t.Run(name, func(t *testing.T) {
+			lvl := zap.NewAtomicLevel()
+			lvl.SetLevel(zapcore.InfoLevel)
+
+			ts := httptest.NewServer(lvl)
+			defer ts.Close()
+
+			req, err := http.NewRequest(test.Method, ts.URL, strings.NewReader(test.Body))
```

Comment:
can we name the test server srv or server? 

```suggestion
			server := httptest.NewServer(lvl)
			defer server.Close()

			req, err := http.NewRequest(test.Method, server.URL, strings.NewReader(test.Body))
```

Reply:
Done